helix-core/src/movement.rs

```diff
@@ -58,80 +104,55 @@ pub fn move_vertically(
 
     let pos = pos_at_coords(text, Position::new(new_line, new_col));
 
-    let mut range = Range::new(if extend { range.anchor } else { pos }, pos);
+    let anchor = match behaviour {
+        SelectionBehaviour::Extend => range.anchor,
+        SelectionBehaviour::Displace => pos,
+    };
+
+    let mut range = Range::new(anchor, pos);
     range.horiz = Some(horiz);
     range
 }
 
-pub fn move_next_word_start(slice: RopeSlice, mut begin: usize, count: usize) -> Option<Range> {
-    let mut end = begin;
-
-    for _ in 0..count {
-        if begin + 1 == slice.len_chars() {
-            return None;
-        }
-
-        let mut ch = slice.char(begin);
-        let next = slice.char(begin + 1);
-
-        // if we're at the end of a word, or on whitespce right before new one
-        if categorize(ch) != categorize(next) {
-            begin += 1;
-        }
+pub fn move_next_word_start(slice: RopeSlice, range: Range, count: usize) -> Range {
```

Comment:
This is the only of the "word movement" functions that is finished; the other two would be modeled after this.